Sources/core/objects/Tables.cpp

```diff
@@ -0,0 +1,279 @@
+            {
+            }
+            else
+            {
+            }
+        }
+    }
+    else if (up_ + down_ + left_ + right_ == 2)
+    {
+        if (up_ + down_ == 2)
+        {
+            SetState(material_ + "_table_d2");
+            Rotate(D_UP);
+        }
+        else if (up_ + left_ == 2)
+        {
+            if (upleft_)
+            {
+                SetState(material_ + "_table_d2_c_r");
+                Rotate(D_UP);
+            }
+            else
+            {
+                SetState(material_ + "_table_d2_c_f");
+                Rotate(D_UP);
+            }
+        }
+        else if (up_ + right_ == 2)
+        {
+            if (upright_)
+            {
+                SetState(material_ + "_table_d2_c_r");
+                Rotate(D_RIGHT);
+            }
+            else
+            {
+                SetState(material_ + "_table_d2_c_f");
+                Rotate(D_RIGHT);
+            }
+        }
+        else if (down_ + left_ == 2)
+        {
+            if (downleft_)
+            {
+                SetState(material_ + "_table_d2_c_r");
+                Rotate(D_DOWN);
+            }
+            else
+            {
+                SetState(material_ + "_table_d2_c_f");
+                Rotate(D_DOWN);
+            }
+        }
+        else if (down_ + right_ == 2)
+        {
+            if (downright_)
+            {
+                SetState(material_ + "_table_d2_c_r");
+                Rotate(D_LEFT);
+            }
+            else
+            {
+                SetState(material_ + "_table_d2_c_f");
+                Rotate(D_LEFT);
+            }
+        }
+        else if (right_ + left_ == 2)
+        {
+            SetState(material_ + "_table_d2");
+            Rotate(D_LEFT);
+        }
+    }
+    else if (up_ + down_ + left_ + right_ == 1)
+    {
+        SetState(material_ + "_table_d1");
+        if (up_ == 1)
+        {
+            Rotate(D_UP);
+        }
+        if (down_ == 1)
+        {
+            Rotate(D_DOWN);
+        }
+        if (left_ == 1)
+        {
+            Rotate(D_LEFT);
+        }
+        if (right_ == 1)
+        {
+            Rotate(D_RIGHT);
+        }
+    }
+    else if (up_ + down_ + left_ + right_ == 0)
+    {
+        SetState(material_ + "_table_d0");
+    }
+}
+void Tables::SetTable(int direction, bool table)
+{
+    switch (direction)
+    {
+    case 1 : up_ = table;
+        break;  
+    case 2 : down_ = table;
+        break;
+    case 3 : left_ = table;
+        break;
+    case 4 : right_ = table;
+        break;
+    case 5 : upleft_ = table;
+        break;  
+    case 6 : downleft_ = table;
+        break;
+    case 7 : upright_ = table;
+        break;
+    case 8 : downright_ = table;
+        break;
+    }
+}
+void Tables::AttackBy(IdPtr<Item> item)
+{
+    if (IdPtr<Human> human= item->GetOwner())
+    {
+        if(auto r_item = dynamic_cast<HumanInterface*>(human->GetInterface())->GetActiveHand().Get())
+            {
```

Comment:
It is an issue with `Human` and `HumanInterface`. For a while just add non-virtual function `GetHumanInterface()` in the `Human` class. It should return pointer to `HumanInterface`, and `Human` implementation of `GetInterface` should use `GetHumanInterface`. Use `GetHumanInterface` here. 
But actually in the future it will be just something like that:
```
item->GetOwner()->RemoveItem(item);
GetOwner()->AddItem(item);
```
It won't work now because `Human` not handling `RemoveItem` function -> I will fix it later.